### ISSUE TITLE
Is there a better way to break user idle?

### ISSUE THREAD

**Issue by paulshapiro** [mymonero/mymonero-app-android#8]
Is there a more global way of detecting all screen input to break user idle than each activity intercepting touches via `dispatchTouchEvent`?

**Comment by KillerInk** [comment]
hi, i think you are looking for this:

> Called whenever a key, touch, or trackball event is dispatched to the activity. Implement this method if you wish to know that the user has interacted with the device in some way while your activity is running.

https://developer.android.com/reference/android/app/Activity.html#onUserInteraction()

**Comment by paulshapiro** [comment]
Yesss 🌠

**Comment by paulshapiro** [comment]
Actually.. that's still per-activity right?

**Comment by KillerInk** [comment]
yes, its still per activity. you have to forward it to the class it need.

**Comment by KillerInk** [comment]
simple java sample how it could work

```
private Handler userIdleHandler =new Handler();

    Runnable userIdleExecuter =new Runnable() {
        @Override
        public void run() {
            setUserIsIdle();
        }
    };

    @Override
    public void onUserInteraction() {
        super.onUserInteraction();
        //remove old executer
        userIdleHandler.removeCallbacks(userIdleExecuter);
        //log user out after 10sec
        userIdleHandler.postDelayed(userIdleExecuter,10*1000);
    }
```

**Comment by KillerInk** [comment]
There is no global way to get the userinput.
You should see the Activity as the main entry.
You will face simliar problems when you need the ActivityContext.
If you need different Uis, its better to use Fragments and load it to the activity,instead of using different activitys.
